### ISSUE TITLE
2.8.11 bom manages com.fasterxml.jackson.dataformat:jackson-dataformat-ion:2.8.11 which does not exist

### ISSUE THREAD

**Issue by wilkinsona** [FasterXML/jackson-bom#10]
As far as I can tell, [this dependency management](https://github.com/FasterXML/jackson-bom/blob/jackson-bom-2.8.11/pom.xml#L64-L68) should not be in the 2.8.x bom as the module's new in 2.9.

**Comment by vdotjansen** [comment]
It seems that 2.8.11 was not released, although there was a 2.8.10 release of it http://central.maven.org/maven2/com/fasterxml/jackson/dataformat/jackson-dataformat-ion/2.8.10/jackson-dataformat-ion-2.8.10.pom

**Comment by tan9** [comment]
@cowtowncoder , for project those adheres to schematic versioning, will there be any *FIX* in this issue? Or just as what @vdotjansen said, it is recommended to upgrade to 2.9.x for all 2.8.x users?

**Comment by cowtowncoder** [comment]
Does this cause a failure? I would have expected that including versions would only add metadata and have no effect unless someone actually tries to add explicit dependency?
Or is there some tool that uses this metadata

@tan9 I had unfortunately missed this issue, so I was unaware of the problem. And even now I want to understand specific effect. But no, upgrade to 2.9 should not be required.

@wilkinsona Official release in 2.9, but versions released for 2.8 as well (http://mvnrepository.com/artifact/com.fasterxml.jackson.dataformat/jackson-dataformat-ion). Whether it should be included in 2.8 bom is debatable, but since it is in I'll need to push 2.8.11.

**Comment by wilkinsona** [comment]
@cowtowncoder It'll only cause a failure if you're using the 2.8.x bom and try to use `com.fasterxml.jackson.dataformat:jackson-dataformat-ion`. If you're not using the module the problem's benign.

**Comment by cowtowncoder** [comment]
@wilkinsona Ok. I'll go ahead and release 2.8.11 version, regardless (since earlier versions are out). Just curious as to specific failure mode.

**Comment by cowtowncoder** [comment]
Hmmh. Actually, I will have to take that back. Releasing of 2.8.11 of `jackson-dataformat-ion` is unfortunately not practical: repo was stand-alone for a brief period of time, before officially added in 2.9 in `jackson-dataformats-binary` repo (which contained other 3 codecs for 2.8).
I deleted stand-alone repository when deciding that 2.8.11 is the last 2.8 full release, thinking it would not be needed.

So: there is no 2.8.11 for `jackson-dataformat-ion`, nor will there be. `2.8.10` works, if anyone wants to use it with 2.8. But that requires override in dependency management section.

But I would recommend upgrade to 2.9 if actually using the Ion module: this is the major version in which it was officially released. If 2.8 is needed for backwards compatibility with other frameworks it does work, just requires manual version override.

**Comment by tan9** [comment]
@cowtowncoder maybe we can have a corrected `jackson-bom` which linked to `jackson-dataformat-ion`:`2.8.10`. Say something like `2.8.11.20180404` just like https://github.com/FasterXML/jackson-bom/compare/jackson-bom-2.8.11...jackson-bom-2.8.11.20180217?

**Comment by cowtowncoder** [comment]
@tan9 I can update `2.8` branch, so if there is another bom version (possibly but not very likely), change would get deployed.

But at this point I am not quite sure as to how useful the fix is: Ion module is not widely used, as far as I know, so is there real deployment problem? I could see a significant problem if there was deployed code that used dependency versions as metadata, and tried automatic processing.
As things are, however, this seems to more of a theoretical problem -- I would like to see actual real concrete issue before using time on it.

**Comment by tan9** [comment]
@cowtowncoder I never heard of `ion` before.
My use case is that we have a server farm physically disconnected from the internet. And I have to manually move in libraries onto a private maven repository. We decided to sync all Jackson's modules and dependencies by walking through `jackson-bom`'s `depenedncyManagement` section. Here comes the problem!

I ended up by overriding the version of `jackson-dataformat-ion` by hand. And I think that I am not the only one tried to resolve dependencies from BOM. Otherwise, this issue won't be there, especially it was opened from the Spring Boot team.
